packages/ui/client/components/explorer/ExplorerItem.vue

```diff
@@ -68,12 +69,22 @@ function toggleOpen() {
 }
 
 async function onRun(task: Task) {
+  if (state === 'todo') {
+    return
+  }
+
   onItemClick?.(task)
   if (coverageEnabled.value) {
     disableCoverage.value = true
     await nextTick()
   }
-  await runFiles([task.file])
+
+  if (type === 'file') {
```

Comment:
Can't we also run folders?

Reply:
we don't have folders structure in the ui, we only have the tasks from the rpc call (collect and getTestFiles/getFiles): if the new api has a new folder type we'll need to change the ui